ch5/exercise5-9/expand_test.go

```diff
@@ -0,0 +1,25 @@
+package expand 
+
+import "testing"
+
+
+func TestExpand(t *testing.T) {
+	var tests = []struct {
+		input string
+		output string 
+	}{
+		{"$foo $bar $hello$hi", "foofoo barbar hello$hihello$hi"}, 
+		{"testing $123 testing$", "testing 123123 testing$"}, 
+		{"go $go $go asd $fghjkl", "go gogo gogo asd fghjklfghjkl"}, 
+		{"$ $ $", "$$ $$ $$"}, 
+		{"", ""}, 
+		{"$helloworld", "helloworldhelloworld"},
+	}
+
+	for _, test := range tests {
+		if got := expand(test.input, f); got != test.output {
+			t.Errorf("Expected: %q = %v", test.input, test.output)
+			t.Errorf("Actual: %q = %v", test.input, got)
```

Comment:
Not explicitly doing that, but I will implement it. I actually like the idea of sub tests, it follows a similar pattern of the library I was using for testing when I was working in Ruby. It will def help with handling failures.

Reply:
I get both outputs when I use 2  `t.Errorf`'s

```
--- FAIL: TestExpand (0.00s)
    expand_test.go:25: Expected: "$foo $bar $hello$hi" = fooisr barisr hello$hiis
    expand_test.go:26: Actual: "$foo $bar $hello$hi" = fooisr barisr hello$hiisr
```

Reply:
> I get both outputs when I use 2 t.Errorf's
Sorry for leading you astray.  TIL.